src/HttpMethodsClient.php

```diff
@@ -11,6 +11,14 @@
  *
  * Use this interface when you do not have PSR-7 RequestInterface instances available.
  *
+ * Implementations should be configured in their constructor. For runtime behaviour changes, there
+ * is an $options parameter. It SHOULD be validated by the client and an exception thrown if
+ * unknown options are passed.
```

Comment:
I don't agree, we also need this settings to be set in the request call,

As an exemple, for our library (docker-php) we need to have this settings per request, as i want to have a logs response in a async manner (it's a chunked reponse which is populated from consecutive actions and stay alive "forever"), but we also want to have the stop / start calls in a sync manner (to be sure that it is correctly stop / start before doing anything else).

But i agree with @dbu and we should only inject these options when we know we use a specific client.

Another approach would be to have a "namespaced" option name for each adapter, where all the configuration under this namespace should only be handle by the adapter

```php
// Global / Interoperable option
$options['timeout'] = 30;
// Specific option for socket adapter
$options['socket-adapter'] = [
    'specific-otption-for-socket' => 'value'
];
// Specific option for guzzle6 adapter
$options['guzzle6-adapter'] = [
    'specific-otption-for-guzzle6' => 'value'
];
```